### DISCUSSION TITLE
[Q&A] Issue with netInventory for switches with snmpV3

### DISCUSSION THREAD

**Discussion by waaalex** [glpi-project/glpi-agent#477] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

HEllo, 
this case have been opened :
https://github.com/glpi-project/glpi-inventory-plugin/issues/405

When i launch net inventory for my switches it does not work when switches are in snmpV3 only.

For working, i have to disable "snmpv3 only" option and put "public" community in snmpV2 on my switches.
Even if in IP scope i only have SNMP3, glpiinventory needs snmpv2.

```
[Fri Sep  8 14:10:06 2023][info] running task NetInventory
[Fri Sep  8 14:10:24 2023][error] [thread 127] #06, [thread 127] SNMP communication error: no response from host 10.0.3.5
[Fri Sep  8 14:10:24 2023][error] [thread 123] #10, [thread 123] SNMP communication error: no response from host 10.0.3.7
.................
[Fri Sep  8 14:10:44 2023][info] running task Deploy
```

Another snmpV2 name does not work and with this command line on an agent : 

`glpi-netinventory.bat --host 10.0.3.3 --credentials version:3,username:snmpV3agent,authprotocol:md5,authpassword:A,privprotocol:aes,privpassword:A --debug`

It works.
Can you help me?
Thanks.Regards.

**Comment by g-bougard** [comment]
Hi @waaalex 

can you confirm a point: did you first discover the devices with a netdiscovery task or did you configure the credential to use on the computer itself ?

One test you can run it to start glpi-agent from the commandline with `--debug --debug` options after the task has been forced in glpi-inventory plugin. This should show you all the exchanges between agent & glpi server and especially, you should be able to verify what credentials the server provides to the agent. To debug this case, I advise you to create an ip range with only one ip to inventory. Here is the command to run just after the task is forced in the plugin and you must run is from an administrative console and from the agent installation folder:
```
glpi-agent --logger=stderr --debug --debug --force --task=netinventory --server=http://you-server/marketplace/glpiinventory
```

You should see something like:
```
<?xml version="1.0"?>
<REPLY><PROLOG_FREQ>24</PROLOG_FREQ><RESPONSE>SEND</RESPONSE><OPTION><NAME>SNMPQUERY</NAME><PARAM THREADS_QUERY="10" TIMEOUT="15" PID="16"/><DEVICE TYPE="NETWORKING" ID="8" IP="192.168.1.253" AUTHSNMP_ID="2"/><AUTHENTICATION><ID>1</ID><VERSION>1</VERSION><COMMUNITY>public</COMMUNITY></AUTHENTICATION><AUTHENTICATION><ID>2</ID><VERSION>2c</VERSION><COMMUNITY>public</COMMUNITY></AUTHENTICATION></OPTION></REPLY>
```
This is the xml sent by the server, here with a basic sample to use snmp v1 & snmp v2 "public" community. You should check `AUTHENTICATION` node contains the expected credentials.

**Reply by g-bougard** [reply, marked as answer]
Hi @waaalex 

I deleted your post as I didn't ask you to share the XML because it can disclose your snmp credentials.

Anyway I checked it myself. I see you're still using a SNMPv2 or SNMPv1 "public" community credential for your ranges. If you disabled it on your devices, the error is normal. You must discover the devices with the SNMPv3 credentials so the netinventory task will be configured with it.

**Reply by waaalex** [reply]
Ok I will check that as soon as possible. Thanks.

**Reply by waaalex** [reply]
I have deleted two switches, netdiscovery has worked and netinventory is ok for ALL switches.
The issue seems to be netinventory that didn't updated switches on database.
I have updated case.
Thank you very much.
